pytorch_lightning/trainer/training_loop.py

```diff
@@ -606,9 +587,6 @@ def run_training_epoch(self):
             self.num_optimizers
         )
 
-        # when no val loop is present or fast-dev-run still need to call checkpoints
-        self.check_checkpoint_callback(not (should_check_val or is_overridden('validation_step', model)))
```

Comment:
you can just fix your usecase by adding `sum(self.trainer.num_val_batches) == 0` here. working on a PR https://github.com/PyTorchLightning/pytorch-lightning/pull/5208 fixing more issues there.

Reply:
@rohitgr7, does it make sense to close the issue in #5208, not here? I'm fine with closing the PR if a nicer solution is proposed

Reply:
I'd suggest yes since doing a bit of a refactor there to fix more issues. Your use-case is already fixed there. Mind check if it works for you??